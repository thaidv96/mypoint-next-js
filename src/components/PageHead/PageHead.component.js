import React from "react";
import PropTypes from "prop-types";
import Head from "next/head";
import reset from "~/theme/reset.styles";
import transitionStyles from "~/theme/page-transitions.styles";

const PageHead = ({ title }) => (
  <div className="PageHead">
    <Head>
      <meta charset="UTF-8" />
      <title>Homepage</title>
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <link rel="stylesheet" href="/static/dist/css/bootstrap.min.css" />
      <link rel="stylesheet" href="/static/dist/css/fonts.css" />
      <link rel="stylesheet" href="/static/dist/css/all.min.css" />
      <link rel="stylesheet" href="/static/dist/css/owl.carousel.min.css" />
      <link
        rel="stylesheet"
        href="/static/dist/css/owl.theme.default.min.css"
      />
      <link rel="stylesheet" href="/static/dist/css/style.css" />
      <link rel="stylesheet" href="/static/dist/css/style_responsice.css" />

      <script
        type="text/javascript"
        src="/static/dist/js/jquery-3.4.1.min.js"
      ></script>
      <script
        type="text/javascript"
        src="/static/dist/js/bootstrap.min.js"
      ></script>
      <script
        type="text/javascript"
        src="/static/dist/js/owl.carousel.min.js"
      ></script>
      {/* <title>{title}</title>

      <meta charSet="utf-8" />
      <meta name="viewport" content="width=device-width, initial-scale=1.0" />
      <meta name="theme-color" content="#F7F7F7" />
      <meta name="twitter:card" content="summary" />
      <meta name="twitter:site" content="TODO INFO/static/DISTRICTES" />
      <meta name="twitter:image" content="/static/favicons/og-image.jpg" />
      <meta name="twitter:title" content="TODO INFODISTRICTES" />
      <meta name="twitter:app:country" content="PT" />
      <link rel="manifest" href="/static/manifest.json" />
      <link rel="stylesheet" href="/static/dist/css/bootstrap.min.css" />
      <link rel="stylesheet" href="/static/dist/css/fonts.css" />
      <link rel="stylesheet" href="/static/dist/css/all.min.css" />
      <link rel="stylesheet" href="/static/dist/css/owl.carousel.min.css" />
      <link
        rel="stylesheet"
        href="/static/dist/css/owl.theme.default.min.css"
      />
      <link rel="stylesheet" href="/static/dist/css/style.css" />
      <link rel="stylesheet" href="/static/dist/css/style_responsice.css" />
      <script
        type="text/javascript"
        src="/static/dist/js/jquery-3.4.1.min.js"
      ></script>
      <script
        type="text/javascript"
        src="/static/dist/js/bootstrap.min.js"
      ></script>
      <script
        type="text/javascript"
        src="/static/dist/js/owl.carousel.min.js"
      ></script>

      <meta name="msapplication-TileColor" content="#F7F7F7" />
      <meta
        name="msapplication-config"
        content="/static/favicons/browserconfig.xml"
      />
      <meta
        name="msapplication-TileImage"
        content="/static/favicons/ms-icon-144x144.png"
      />
      <meta property="og:image" content="/static/og-image.jpg" />
      <meta property="og:image:width" content="201" />
      <meta property="og:image:height" content="201" />
      <meta property="og:title" content="TODO INFODISTRICTES" />
      <meta property="og:description" content="TODO INFODISTRICTES" />
      <meta property="og:url" content="TODO INFODISTRICTES" />
      <meta httpEquiv="X-UA-Compatible" content="IE=edge" />
      <meta name="HandheldFriendly" content="True" />
      <meta name="MobileOptimized" content="320" /> */}
    </Head>
    {/* <style jsx global>
      {reset}
    </style>
    <style jsx global>
      {transitionStyles}
    </style> */}
  </div>
);

PageHead.propTypes = {
  title: PropTypes.string,
};

PageHead.defaultProps = {
  title: "TODO DEFAULT TITLE",
};

export default PageHead;
