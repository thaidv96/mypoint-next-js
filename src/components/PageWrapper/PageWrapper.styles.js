import css from 'styled-jsx/css'

export default css`
  .PageWrapper {
    width: 100%;
    min-height: 100vh;
    height: 100%;
    position: relative;
    margin: 0;
  }
  .gradient {
    width: 100%;
    height: 100%;
    min-height: 100vh;
    position: absolute;
    margin: 0;
  }
`
