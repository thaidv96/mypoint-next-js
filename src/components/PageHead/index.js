import PageHead from './PageHead.component'

export default PageHead
