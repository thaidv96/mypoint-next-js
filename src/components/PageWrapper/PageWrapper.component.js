import React from "react";
import PropTypes from "prop-types";
// import TYPES from '~/constants/types'
import "~/utils/polyfills";
import PageHead from "~/components/PageHead";
import Loader from "~/components/Loader";

import styles from "./PageWrapper.styles";

const PageWrapper = ({ children, title }) => (
  <div className="PageWrapper">
    <Loader />
    <PageHead title={title} />
    {children}
    {/* <style jsx>{styles}</style> */}
  </div>
);

PageWrapper.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.func,
  ]),
  title: PropTypes.string,
};

PageWrapper.defaultProps = {
  children: null,
  title: "InfoDistrictes",
};

export default PageWrapper;
