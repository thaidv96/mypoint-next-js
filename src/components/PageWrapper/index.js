import PageWrapper from './PageWrapper.component'

export default PageWrapper
