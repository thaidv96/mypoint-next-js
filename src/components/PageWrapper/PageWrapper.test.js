/* eslint-env jest */
/* global window */
import React from 'react'
import { shallow } from 'enzyme'
import PageHead from '~/components/PageHead'
import PageWrapper from './PageWrapper.component'


describe('Title', () => {
  const baseProps = {
    title: 'hey! a title?',
  }

  const wrapperMobile = shallow(
    <PageWrapper
      {...baseProps}
      isMobile
    >
      <div className="child-one" />
    </PageWrapper>
  )

  const wrapperDesktop = shallow(
    <PageWrapper {...baseProps} >
      <div className="child-two" />
    </PageWrapper>
  )


  it('matches its snapshots', () => {
    expect(wrapperMobile).toMatchSnapshot('mobile')
    expect(wrapperDesktop).toMatchSnapshot('desktop')
  })

  it('renders its kids', () => {
    const childOneExists = wrapperMobile
      .find('.child-one')
      .exists()

    expect(childOneExists).toBe(true)

    const childTwoExists = wrapperDesktop
      .find('.child-two')
      .exists()

    expect(childTwoExists).toBe(true)
  })

  it('passes the correct title prop to PageHead', () => {
    const pageHeadTitle = wrapperDesktop
      .find(PageHead)
      .props()
      .title

    expect(pageHeadTitle).toBe('TODO INFODISTRICTES')
  })

})
