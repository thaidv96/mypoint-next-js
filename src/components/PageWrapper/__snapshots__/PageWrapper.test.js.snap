// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title matches its snapshots: desktop 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageWrapper
    title="hey! a title?"
  >
    <div
      className="child-two"
    />
  </PageWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Loader) />,
        <PageHead
          title="TODO INFODISTRICTES"
        />,
        <div
          className="child-two"
        />,
        <JSXStyle
          css=".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}"
          styleId="3336110975"
        />,
      ],
      "className": "jsx-3336110975 PageWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "TODO INFODISTRICTES",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "child-two",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}",
          "styleId": "3336110975",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Loader) />,
          <PageHead
            title="TODO INFODISTRICTES"
          />,
          <div
            className="child-two"
          />,
          <JSXStyle
            css=".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}"
            styleId="3336110975"
          />,
        ],
        "className": "jsx-3336110975 PageWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "TODO INFODISTRICTES",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "child-two",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}",
            "styleId": "3336110975",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;

exports[`Title matches its snapshots: mobile 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageWrapper
    isMobile={true}
    title="hey! a title?"
  >
    <div
      className="child-one"
    />
  </PageWrapper>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Loader) />,
        <PageHead
          title="TODO INFODISTRICTES"
        />,
        <div
          className="child-one"
        />,
        <JSXStyle
          css=".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}"
          styleId="3336110975"
        />,
      ],
      "className": "jsx-3336110975 PageWrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "title": "TODO INFODISTRICTES",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "child-one",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": ".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}",
          "styleId": "3336110975",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Loader) />,
          <PageHead
            title="TODO INFODISTRICTES"
          />,
          <div
            className="child-one"
          />,
          <JSXStyle
            css=".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}"
            styleId="3336110975"
          />,
        ],
        "className": "jsx-3336110975 PageWrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "title": "TODO INFODISTRICTES",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "child-one",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": ".PageWrapper.jsx-3336110975{width:100%;min-height:100vh;height:100%;position:relative;margin:0;}.gradient.jsx-3336110975{width:100%;height:100%;min-height:100vh;position:absolute;margin:0;}",
            "styleId": "3336110975",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
